### PR TITLE
[v2.6] reload rke2/k3s config on kdm refresh action

### PR DESCRIPTION
**Pull request by kinarashah**
https://github.com/rancher/rancher/issues/34473

### CODE REVIEW

File: pkg/api/norman/customization/kontainerdriver/actionhandler.go [hunk 1]
```diff
@@ -81,20 +80,8 @@ func (a ActionHandler) refresh(apiContext *types.APIContext) error {
...
-	setting, err := a.MetadataHandler.SettingLister.Get("", rkeMetadataConfig)
-	if err != nil {
-		return err
-	}
-
-	if setting.Annotations == nil {
-		setting.Annotations = make(map[string]string)
-	}
-
-	setting.Annotations[forceRefreshAnnotation] = strconv.FormatInt(time.Now().Unix(), 10)
```
Comment (kinarashah) [inline pkg/api/norman/customization/kontainerdriver/actionhandler.go:93]: `forceRefreshAnnotation` is no longer used anywhere

Comment (StrongMonkey) [reply pkg/api/norman/customization/kontainerdriver/actionhandler.go:93]: Do you know how/why it was used before? Is this a problem in 2.5?

**Comment by StrongMonkey** [comment]
Since that action is only for refreshing KDM that is used in rke1, should we really include refreshing k3s/rke2 into KDM or make it a separate api for it? I am open for that discussion.

File: pkg/api/norman/customization/kontainerdriver/actionhandler.go (hunk 1 above)
Comment (kinarashah) [reply pkg/api/norman/customization/kontainerdriver/actionhandler.go:93]: It was used earlier to refresh k3s metadata based on this annotation, the logic still exists in 2.5 but no longer in 2.6 https://github.com/rancher/rancher/blob/release/v2.5/pkg/controllers/managementapi/k3smetadata/settings_handler.go#L66

**Comment by kinarashah** [comment]
@StrongMonkey This is a good question and I kept it in the same action because: 
- we haven't separated settings for rke1 and rke2, the url and interval is still read from the same v3 setting `rke-metadata-config`
- Action `refresh` continues to exist on v3/kontainerdriver schema, unrelated to the underlying cluster type 
- would we weird to have 2 different actions for user to go and refresh when the same setting interval works for both handlers

File: pkg/channelserver/channelserver.go [hunk 2]
```diff
@@ -61,12 +62,21 @@ func (d *DynamicInterval) Wait(ctx context.Context) bool {
 				return true
 			}
 			continue
+		case msg := <-action:
+			if msg == "refresh" {
```
Comment (StrongMonkey) [inline pkg/channelserver/channelserver.go:66]: Nit: use constant for `refresh` as it is used twice.

**Review by StrongMonkey — APPROVED** [review]

File: pkg/channelserver/channelserver.go (hunk 2 above)
Comment (kinarashah) [reply pkg/channelserver/channelserver.go]: updated

**Review by dramich — DISMISSED** [review]

**Review by StrongMonkey — APPROVED** [review]

**Comment by kinarashah** [comment]
Pushed changes again to rebase, no other code change. @StrongMonkey could you re-approve?

**Review by StrongMonkey — APPROVED** [review]

**Pull request merged**